c7n/schema.py

```diff
@@ -578,6 +579,83 @@ def resource_vocabulary(cloud_name=None, qualify_name=True, aliases=True):
     return vocabulary
 
 
+class ExpandedSchemaMeta(type):
```

Comment:
I'm a little wary of a metaclass, it may be appropriate, there are some simpler viable alternatives like a shape_schema function, that takes type_schema metadata and shape. that would allow for much control if there are multiple shapes and placement in the resulting jsonschema. for many shape validations we typically have to do some massaging of the inputs, this the massage is specific to the id parameter. in some cases we're not modeling the full request shape, but an internal shape, and a function approach rather then class metadata maybe able to better model that with more control where it appears in the schema. its worth a discussion at the next community meeting.